main_test.go

```diff
@@ -39,3 +39,10 @@ func TestYAMLParse(t *testing.T) {
 	assert.Contains(t, java.Environment, "ELASTIC_APM_LOG_LEVEL")
 	assert.Contains(t, java.Environment, "ELASTIC_APM_PROFILING_INFERRED_SPANS_ENABLED")
 }
+
+func TestYAMLParseBad(t *testing.T) {
+	p := "testdata/bad.yaml"
+	cfg, err := parseConfig(p)
+	require.Error(t, err)
```

Comment:
Should we check the specific error message? I tend to find doing so helps ensure error messages remain readable over time.